pyanaconda/core/configuration/storage.py

```diff
@@ -142,3 +123,78 @@ def luks_version(self):
 
         return value
+
+    @property
+    def default_partitioning(self):
+        """Default partitioning.
+
+        Returns a list of dictionaries with mount point attributes.
+        The name of the mount point is represented by the attribute
+        'name' in the dictionary representation.
+
+        Valid attributes:
+
+            name       The name of the mount point.
+            size       The size of the mount point.
+            min        The size will grow from min size to max size.
+            max        The max size is unlimited by default.
+            free       The required available space.
+            encrypted  The mount point will be encrypted.
+
+        :return: a list of dictionaries with mount point attributes
+        """
+        return self._get_option("default_partitioning", self._convert_partitioning)
+
+    def _convert_partitioning(self, value):
+        """Convert a partitioning string into a list of dictionaries."""
+        return list(map(self._convert_partitioning_line, value.strip().split("\n")))
+
+    @classmethod
+    def _convert_partitioning_line(cls, line):
+        """Convert a partitioning line into a dictionary."""
+        # Parse the line.
+        name, attributes = cls._split_string(line)
+
+        attributes = attributes.strip("()").split(",")
+        attributes = map(cls._split_string, filter(None, attributes))
+
+        # Generate the dictionary.
+        attrs = {"name": name}
+
+        for name, value in attributes:
+            if not value and name in ("encrypted",):
+                # Handle a boolean attribute.
+                attrs[name] = True
+            elif value and name in ("size", "min", "max", "free"):
+                # Handle a size attribute.
+                attrs[name] = Size(value)
+            else:
+                # Handle an invalid attribute.
+                raise ValueError("Invalid attribute: " + name)
+
+        # Validate the dictionary.
+        cls._validate_mount_point_attributes(attrs)
+
+        return attrs
+
+    @staticmethod
+    def _validate_mount_point_attributes(attrs):
+        """Validate the dictionary with mount point attributes."""
+        if not attrs.get("name"):
+            raise ValueError("The mount point is not specified.")
+
+        if attrs.get("name") != "swap" and not attrs.get("name").startswith("/"):
+            raise ValueError("The mount point is not valid.")
+
+        if attrs.get("size") and attrs.get("min"):
+            raise ValueError("Only one of the attributes 'size' and 'min' can be set.")
+
+        if attrs.get("max") and not attrs.get("min"):
+            raise ValueError("The attribute 'max' cannot be set without 'min'.")
+
+    @staticmethod
+    def _split_string(value):
+        """Split the given value into two strings."""
+        first, *second = value.strip().split(maxsplit=1)
+        second = "".join(second)
+        return first, second
```

Comment:
The combination of `maxsplit` and `*second` to avoid problems during parsing items without values (encrypted) is really confusing. Could you please solve this not that elegant way but pythonic way, by using exceptions here?

Reply:
Fixed.